### PR TITLE
Channel and GoGlow Contact tweaks

### PR DESCRIPTION
**Pull request by rowanseymour**
* Add some extra fields to goflow/editor serialization that we need in goflow for picking channels
* Replace contact.channel in goflow serialization with URN query params for each URN's channel
* Move Yo constants into the Yo type
* Remove the never-used (?) Dummy channel type

### CODE REVIEW

File: temba/channels/models.py [hunk 1]
```diff
@@ -304,16 +302,20 @@ class Channel(TembaModel):
 
     DEFAULT_ROLE = ROLE_SEND + ROLE_RECEIVE
 
+    ROLE_CONFIG = {
+        ROLE_SEND: 'send',
+        ROLE_RECEIVE: 'receive',
+        ROLE_CALL: 'call',
+        ROLE_ANSWER: 'answer',
+        ROLE_USSD: 'ussd',
+    }
```
Comment (rowanseymour) [inline temba/channels/models.py]: for goflow serialization we always use human readable constants

File: temba/channels/models.py [hunk 2]
```diff
@@ -244,7 +243,6 @@ def __str__(self):
 @six.python_2_unicode_compatible
 class Channel(TembaModel):
     TYPE_ANDROID = 'A'
-    TYPE_DUMMY = 'DM'
```
Comment (rowanseymour) [inline temba/channels/models.py]: don't think we've ever created one of these?

File: temba/utils/goflow/serialize.py [hunk 3]
```diff
@@ -29,13 +29,25 @@ def serialize_flow(flow, strip_ui=True):
 
 
 def serialize_channel(channel):
-    return {
+    from temba.channels.models import Channel
+
+    serialized = {
         'uuid': str(channel.uuid),
         'name': six.text_type(channel.get_name()),
-        'type': channel.channel_type,
```
Comment (rowanseymour) [inline temba/utils/goflow/serialize.py]: unless we identify I need for the editor and flowserver to know about these type codes it seems preferable to hide them for now

File: temba/contacts/models.py [hunk 4]
```diff
@@ -106,10 +107,14 @@ def from_parts(cls, scheme, path, display=None):
...
+
+        if query:
+            urn += '?%s' % urlencode(query)
         if display:
-            return '%s:%s#%s' % (scheme, path, display)
-        else:
-            return '%s:%s' % (scheme, path)
+            urn += '#%s' % display
+
+        return urn
```
Comment (nicpottier) [inline temba/contacts/models.py]: https://stackimpact.com/blog/practical-golang-benchmarks/#string-concatenation

Comment (nicpottier) [reply temba/contacts/models.py]: Oops, this is python!

File: temba/utils/goflow/serialize.py (hunk 3 above)
Comment (nicpottier) [reply temba/utils/goflow/serialize.py]: 👍

File: temba/channels/models.py (hunk 2 above)
Comment (ericnewcomer) [reply temba/channels/models.py]: don't see where we would, maybe it was part of that spam feature we had way back when?

Comment (rowanseymour) [reply temba/channels/models.py]: maybe - i checked on prod and didn't find any

**Pull request merged**